### PR TITLE
chore: Remove unused hooks

### PR DESCRIPTION
**Pull request by RabbitDoge**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: src/contexts/RefreshContext.tsx [hunk 1]
```diff
@@ -1,34 +1,52 @@
...
+    window.addEventListener('visibilitychange', onVisibilityChange)
+
+    return () => {
+      window.removeEventListener('visibilitychange', onVisibilityChange)
+    }
+  }, [])
+
+  return isBrowserTabActiveRef
+}
+
```
Comment (hachiojidev) [inline src/contexts/RefreshContext.tsx:26]: I think this should be its own hook. You can use this other places where you want to detect if the browser tab is hidden or not.

File: src/utils/calls/estimateGas.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export const estimateGas = async (contract: Contract, methodName: string, methodArgs: any[]) => {
+  if (!contract[methodName]) {
+    throw new Error(`Method ${methodName} doesn't exist on ${contract.address}`)
+  }
+  const rawGasEstimation = await contract.estimateGas[methodName](...methodArgs)
+  // By convention, ethers.BigNumber values are multiplied by 1000 to avoid dealing with real numbers
+  const gasEstimation = rawGasEstimation
+    .mul(ethers.BigNumber.from(10000).add(ethers.BigNumber.from(1000)))
+    .div(ethers.BigNumber.from(10000))
+  return gasEstimation
```
Comment (hachiojidev) [inline src/utils/calls/estimateGas.ts:19]: Didn't you do this in another PR?

File: src/views/Farms/hooks/useApprove.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+import { ethers, Contract } from 'ethers'
+import { useMasterchef } from 'hooks/useContract'
+
+const useApprove = (lpContract: Contract) => {
```
Comment (hachiojidev) [inline src/views/Farms/hooks/useApprove.ts:5]: We should consider renaming this to something like `useApproveLpContract`

File: src/views/Farms/hooks/useHarvest.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { useCallback } from 'react'
+import { harvest } from 'utils/calls'
+import { useMasterchef } from 'hooks/useContract'
+
+const useHarvest = (farmPid: number) => {
```
Comment (hachiojidev) [inline src/views/Farms/hooks/useHarvest.ts:5]: Same here, `useHarvestFarm` or something.

File: src/views/Farms/hooks/useStake.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react'
+import { stake } from 'utils/calls'
+import { useMasterchef } from 'hooks/useContract'
+
+const useStake = (pid: number) => {
```
Comment (hachiojidev) [inline src/views/Farms/hooks/useStake.ts:5]: And here...maybe `useStakeInPool`

File: src/views/Farms/hooks/useUnstake.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { useCallback } from 'react'
+import { unstake } from 'utils/calls'
+import { useMasterchef } from 'hooks/useContract'
+
+const useUnstake = (pid: number) => {
```
Comment (hachiojidev) [inline src/views/Farms/hooks/useUnstake.ts:5]: and...here

File: src/contexts/RefreshContext.tsx (hunk 1 above)
Comment (RabbitDoge) [reply src/contexts/RefreshContext.tsx:26]: Yes but we don't. And I don't see a single use case where we want to use it. If one day we need it elsewhere, we will move it.

File: src/utils/calls/estimateGas.ts (hunk 2 above)
Comment (RabbitDoge) [reply src/utils/calls/estimateGas.ts:19]: just moving to s separate file

Comment (RabbitDoge) [reply src/utils/calls/estimateGas.ts:19]: image

**Review by hachiojidev — APPROVED** [review]
